File: backend/app/schemas/person.py
from datetime import date, datetime
from typing import Optional
from pydantic import BaseModel, validator
from enum import Enum


class ProfessionEnum(str, Enum):
    INGENIERO = "Ingeniero"
    MEDICO = "Médico"
    ABOGADO = "Abogado"
    PROFESOR = "Profesor"
    CONTADOR = "Contador"
    ARQUITECTO = "Arquitecto"
    ENFERMERO = "Enfermero"
    PSICOLOGO = "Psicólogo"
    VETERINARIO = "Veterinario"
    OTRO = "Otro"


class PersonBase(BaseModel):
    first_name: str
    last_name: str
    birth_date: date
    profession: ProfessionEnum
    address: str
    phone: str

    @validator('first_name', 'last_name')
    def validate_names(cls, v):
        if not v or len(v.strip()) < 2:
            raise ValueError('Nombres y apellidos deben tener al menos 2 caracteres')
        return v.strip()

    @validator('phone')
    def validate_phone(cls, v):
        if not v or len(v) < 10:
            raise ValueError('El teléfono debe tener al menos 10 dígitos')
        return v

    @validator('address')
    def validate_address(cls, v):
        if not v or len(v.strip()) < 5:
            raise ValueError('La dirección debe tener al menos 5 caracteres')
        return v.strip()


class PersonCreate(PersonBase):
    pass


class PersonUpdate(PersonBase):
    pass


class PersonInDB(PersonBase):
    id: int
    age: int
    photo_url: Optional[str] = None
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class PersonResponse(PersonInDB):
    pass


class PersonStats(BaseModel):
    total_persons: int
    professions_count: dict
    age_ranges: dict
    monthly_registrations: dict

File: frontend/src/domain/entities/Person.ts
export enum ProfessionEnum {
  INGENIERO = "Ingeniero",
  MEDICO = "Médico",
  ABOGADO = "Abogado",
  PROFESOR = "Profesor",
  CONTADOR = "Contador",
  ARQUITECTO = "Arquitecto",
  ENFERMERO = "Enfermero",
  PSICOLOGO = "Psicólogo",
  VETERINARIO = "Veterinario",
  OTRO = "Otro"
}

export interface Person {
  id: number;
  first_name: string;
  last_name: string;
  birth_date: string; 
  age: number;
  profession_id: number;
  profession_name?: string;
  address: string;
  phone: string;
  photo_url?: string;
  created_at?: string;
  updated_at?: string;
}

export interface CreatePersonRequest {
  first_name: string;
  last_name: string;
  birth_date: string; 
  profession_id: number;
  address: string;
  phone: string;
  photo?: File;
}

export interface UpdatePersonRequest {
  first_name: string;
  last_name: string;
  birth_date: string;
  profession_id: number;
  address: string;
  phone: string;
  photo?: File;
}

export interface PersonFormData {
  first_name: string;
  last_name: string;
  birth_date: string;
  profession_id: number;
  address: string;
  phone: string;
  photo?: File;
}

export interface PersonStats {
  total_persons: number;
  professions_count: Record<string, number>;
  age_ranges: Record<string, number>;
  monthly_registrations: Record<string, number>;
}
